src/buffer_storage.cc

```diff
@@ -35,16 +35,18 @@ namespace detail {
 	void memcpy_strided(const void* source_base_ptr, void* target_base_ptr, size_t elem_size, const cl::sycl::range<3>& source_range,
 	    const cl::sycl::id<3>& source_offset, const cl::sycl::range<3>& target_range, const cl::sycl::id<3>& target_offset,
 	    const cl::sycl::range<3>& copy_range) {
+		const cl::sycl::range<2> source_range_2d{source_range[1], source_range[2]};
```

Comment:
Even though this is arguably more readable, I would propose to remove this entire change set, as the underlying issue has already been fixed in 3ded4d78.